### PR TITLE
Requests

### PR DESCRIPTION
**Pull request by huebnerob**
https://vimean.atlassian.net/browse/TVOS-43

This PR adds some more high-level template requests to our library.  From here, we can probably just add any more as needed.

### CODE REVIEW

**Comment by huebnerob** [comment]
@alfiehanssen @mikew-personal @jasonhawkins Can I get a review from somebody? 🙏

**Comment by jasonhawkins** [comment]
I can review!

**Comment by jasonhawkins** [comment]
I had to update the bridging header to get this to build. It looks like this was a problem before too, but can you change the imports in `VimeoNetworking-Bridging-Header.h` to: 
```ObjC
#import <VimeoNetworking/VIMUser.h>
#import <VimeoNetworking/VIMAccountNew.h>
#import <VimeoNetworking/VIMVideo.h>
```

**Comment by jasonhawkins** [comment]
What's the best way for me to test this? The tvOS project?

**Comment by huebnerob** [comment]
Thanks @jasonhawkins! 

You could mess with making some requests somewhere in the sample project and ensure they return expected results.

**Comment by jasonhawkins** [comment]
I'm getting 400's when I run the sample project just as it is. Is that how it's been/supposed to be?

**Comment by huebnerob** [comment]
It might be, depending on which authentication method is currently commented/uncommented.  Some of the test requests are trying the `/me` endpoint which won't work if we're using client credentials grant.  I'd recommend replacing everything in the app delegate `testEndpoints()` function with whatever you want to try out.

File: VimeoNetworking/VimeoNetworking/Request+Category.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import Foundation
+
+public typealias CategoryRequest = Request<VIMCategory>
+public typealias CategoryListRequest = Request<[VIMCategory]>
+
+public extension Request
+{
+    private static var CategoriesPath: String { return "/categories" }
+    
+    public static func getCategoryRequest(categoryURI categoryURI: String) -> Request
```
Comment (jasonhawkins) [inline VimeoNetworking/VimeoNetworking/Request+Category.swift]: How do you feel about adding documentation for these public methods?

File: VimeoNetworking/VimeoNetworking/Request+Channel.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    
+    public static func queryChannels(query query: String, refinements: VimeoClient.RequestParameters? = nil) -> Request
+    {
+        var parameters = refinements ?? [:]
+        
+        parameters[self.QueryKey] = query
+        
+        return Request(path: self.ChannelsPath, parameters: parameters)
+    }
+}
```
Comment (jasonhawkins) [inline VimeoNetworking/VimeoNetworking/Request+Channel.swift]: Git wants new lines at the end of all of these.

File: VimeoNetworking/VimeoNetworking/Request+User.swift [hunk 3]
```diff
@@ -35,4 +47,46 @@ public extension Request
...
+    
+    public static func getUserFollowersRequest(userURI userURI: String) -> Request
+    {
+        return Request(path: String(format: self.FollowersPathFormat, userURI))
+    }
+    
+    // MARK: - Search
+    
+    
+
```
Comment (jasonhawkins) [inline VimeoNetworking/VimeoNetworking/Request+User.swift]: Extra whitespace.

File: VimeoNetworking/VimeoNetworking/Request+Category.swift [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    {
+        return Request(path: categoryURI)
+    }
+    
+    public static func getCategoriesRequest() -> Request
```
Comment (jasonhawkins) [inline VimeoNetworking/VimeoNetworking/Request+Category.swift]: Do these requests need to be authenticated for them not to fail? Are there any that don't require authentication?

**Comment by jasonhawkins** [comment]
Looks like you'll need to pull in the latest from master here.

File: VimeoNetworking/VimeoNetworking/Request+Channel.swift (hunk 2 above)
Comment (alfiehanssen) [inline VimeoNetworking/VimeoNetworking/Request+Channel.swift]: Is that what that means? Why does it want that? 😨

File: VimeoNetworking/VimeoNetworking/Request.swift [hunk 5]
```diff
@@ -10,20 +10,36 @@ import Foundation
...
 public enum CacheFetchPolicy
 {
-    static let DefaultPolicy: CacheFetchPolicy = .CacheThenNetwork
-    
     case CacheOnly
     case CacheThenNetwork
     case NetworkOnly
     case TryNetworkThenCache
+    
+    static func defaultPolicyForMethod(method: VimeoClient.Method) -> CacheFetchPolicy
```
Comment (alfiehanssen) [inline VimeoNetworking/VimeoNetworking/Request.swift]: oooooh 👍

File: VimeoNetworking/VimeoNetworking/Request+Category.swift (hunk 4 above)
Comment (huebnerob) [inline VimeoNetworking/VimeoNetworking/Request+Category.swift]: Everything needs an access token (client credentials, minimum) but, no, these don't need you to be logged in as a user

**Pull request merged**